src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
I thought maybe we could mix that in automatically for mixins that require it, again might be that it's not possible.

```ts
export function FakeMixin<T extends Constructor<WidgetBase<FakeMixinProperties>>>(base: T): T {
	class FakeMixin extends PropagatePropertyMixin(base) {

        @afterRender
        protected propagateProperties(result: DNode) {
            decorate(result, (result: DNode) => {}, isWNode);
            return result;
        }
    }
    return FakeMixin;
}

export default FakeMixin;
```